backend/corpora/dataset_processing/remaster_cxg_v2.py

```diff
@@ -0,0 +1,215 @@
+    :param dataset_id: The id of the dataset to reprocess.
+    :param cellxgene_bucket: Name of the S3 bucket where the artifact is.
+    :param prefix: If specified, will
+    :param dry_run:
+    :return:
+    """
+
+    # Download the cxg from the bucket
+    if prefix is not None:
+        object_key = f"{prefix}{dataset_id}"
+    else:
+        dataset_path = get_bucket_prefix(dataset_id)
+        object_key = f"{dataset_path}.cxg"
+    path = f"s3://{cellxgene_bucket}/{object_key}/X"
+
+    logger.info(f"Processing dataset at path {path}, dry run {dry_run}")
+
+    local_path = "/cxg"
+    logger.info(f"Downloading {path} to {local_path}/X_old")
+
+    download_command = ["aws", "s3", "sync", path, f"{local_path}/X_old"]
+    # Let errors fail the pipeline
+    subprocess.run(download_command, check=True)
+
+    params = {
+        "kind": "auto",
+        "obs_extent": 256,
+        "var_extent": 256,
+        "cell_order": "row",
+        "tile_order": "row",
+        "capacity": 1024000,
+        "compression": 22,
+        "target_array": "X_new",
+        "source_array": "X_old",
+    }
+
+    executed = compute(cxg=local_path, **params)  # executed is true if a sparse array was upgraded
+    if executed:
+        logger.info(f"Dataset at {path} computed successfully")
+    else:
+        logger.info("Dataset was dense, nothing to be done")
+
+    if not dry_run and executed:
+        for suffix in ["", "c"]:
+            upload_command = ["aws", "s3", "sync", "--delete", f"{local_path}/X_new" + suffix, path + suffix]
+            subprocess.run(upload_command, check=True)
+
+    # Cleanup
+    logger.info("Cleaning up local files")
+    import shutil
+
+    shutil.rmtree(f"{local_path}/X_old")
+    if executed:
+        shutil.rmtree(f"{local_path}/X_new")
+
+
+def compute(**kwargs):
+    """
+    Computes the evolved cxg from the sparse source_array and saves it to the target_array destination
+    :param cxg:
+    :param obs_extent:
+    :param var_extent:
+    :param source_array:
+    :param target_array:
+    :param kind:
+    :param sparse_threshold:
+    :param compression:
+    :param cell_order:
+    :param tile_order:
+    :param capacity:
+    """
+
+    cxg = kwargs["cxg"]
+    obs_extent = kwargs["obs_extent"]
+    var_extent = kwargs["var_extent"]
+    source_array = kwargs["source_array"]
+    target_array = kwargs["target_array"]
+    compression = kwargs["compression"]
+    cell_order = kwargs["cell_order"]
+    tile_order = kwargs["tile_order"]
+    capacity = kwargs["capacity"]
+
+    X_extent = [obs_extent, var_extent]
+
+    ctx = create_fast_ctx(
+        {
+            "py.init_buffer_bytes": 4 * 1024**3,
+            "sm.tile_cache_size": 1 * 1024**3,
+        }
+    )
+
+    st = time.time()
+    with tiledb.scope_ctx(ctx):
+        with tiledb.open(f"{cxg}/{source_array}", "r") as old_X:
+            in_sparse = old_X.schema.sparse
```

Comment:
`is_sparse`

Reply:
This was adapted from `backend/corpora/dataset_processing/remaster_cxg.py` where they used `in_sparse` and `out_sparse`. Here, `out_sparse` always equals True so I deleted that variable.